### ISSUE TITLE
[PROPOSAL] the <Errors> component

### ISSUE THREAD

**Issue by davidkpiano** [davidkpiano/react-redux-form#124]
Even for the simplest cases, displaying error messages can get pretty verbose:

```js
{ userForm.fields.email.touched
  && userForm.fields.email.errors.isRequired
  && <div>Email address is required</div>
}

// ... etc. for other errors
```

What if we had a robust `<Errors>` component to help cut down this verbosity? The above example would become:

```js
<Errors model="user.email" messages={{
  isRequired: 'Email address is required',
  isEmail: 'Must be a valid email address',
}} />
```

**THE SPECIFICS**

The full props for the `<Error>` component would look something like this:

```js
<Errors
  model="user.email" // string or model getter function, just like Field
  messages={{...}} // map - <validation key>: <message> (string or JSX)
  show="touched" // string or array or function that takes field object as arg
  wrapper="div" // string or component class for container
  component="span" // string or component class for each child error
/>
```

The required props are **`model`** and **`messages`**.

Thoughts? I'd like to think this one through before introducing it.

**Comment by lasergoat** [comment]
correction: `show="touched"`

This sounds good. though I'd like to be able to specify a className for the child spans. What if we could pass JSX into the component field.

`component={<span className="blah"></span>}`

`component={<MyElement></MyElement>}`

**Comment by chrisblossom** [comment]
First thought it seems like it could make styles etc complicated. What about just having a showError true/false state per field? 

```dispatch(rrf.showError('userForm.email', true/false));```

If you wanted to expand on that, there could be some config that would set the params to show errors.

```js
const config = {
  showErrors: [touched, !valid, submitFailed],
}
```

**Comment by davidkpiano** [comment]
> First thought it seems like it could make styles etc complicated. What about just having a showError true/false state per field?

I feel like this would make things more complicated, since you'd have to do this for every field and would result in unnecessary rerenders.

> EDIT: Also, the ability to store error messages already exists with the errors field (I think also possible with the validity field as well, just more difficult because of the way valid works.

It's not encouraged, however, because of how validation works. RRF assumes that you won't hard-code error messages in your validation functions, but still gives you the option to have anything truthy (like a string) represented in `.errors`.

Per @lasergoat 's suggestion, I'm thinking along the lines of this for styling:

```js
<Errors model="..."
  component={(msg) => <span className="error">{msg}</span>}
/>
```

which will give a lot of flexibility to how your error messages are displayed.

**Comment by davidkpiano** [comment]
This is now available in `react-redux-form@0.10.0`!
